### PR TITLE
docs: update range query docs

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
## What's Changed in this PR

update range query docs
- new fill behavior
- Range SQL structure
- `first_value`/`last_value` use `ts` as default `order by` 

related pr: https://github.com/GreptimeTeam/greptimedb/pull/3489

## Checklist

- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

Closes https://github.com/GreptimeTeam/docs/issues/832

### CODE REVIEW

File: docs/v0.7/en/reference/sql/range.md [hunk 1]
```diff
@@ -23,15 +26,15 @@ INTERVAL :=  TIME_INTERVAL | ( INTERVAL expr )
    - Strings based on the `PromQL Time Durations` format (eg: `3h`, `1h30m`). Visit the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/querying/basics/#time-durations) for a more detailed description of this format.
    - `Interval` type. To use the `Interval` type, you need to carry parentheses, (for example: `(INTERVAL '1 year 3 hours 20 minutes')`). Visit [Interval](./functions.md#interval) for a more detailed description of this format.
 - `AGGR_FUNCTION(column1, column2,..) RANGE INTERVAL [FILL FILL_OPTION]` is called a Range expression.
-   - `AGGR_FUNCTION(column1, column2,..)` is an aggregate function that represents the expression that needs to be aggregated.
+   - `AGGR_FUNCTION(column1, column2,..)` is an aggregate function that represents the expression that needs to be aggregated. Aggregate functions support the use of `order by` expressions. (Note: `first_value` and `last_value` functions use timestamp as `order by` expression by default)
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:29]: I don't understand what NOTE means. Does that mean the type of the column specified by `order by` must be a timestamp? Does that mean that other aggregate functions can use any type of column as an order by parameter?

File: docs/v0.7/en/reference/sql/range.md [hunk 2]
```diff
@@ -60,74 +63,135 @@ ALIGN '5s' BY (host) FILL PREV;
 | `LINEAR` | Use [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation) to fill the data. If an integer type is filled with `LINEAR`, the variable type of the column will be implicitly converted to a floating point type during calculation |
 |   `X`    |                                           Fill in a constant, the data type of the constant must be consistent with the variable type of the Range expression                                            |
 
+:::tip NOTE
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:66]: A note shouldn't be too long, and please avoid using too many notes.
Here seems to be a special case of the `FILL` option, so please remove the `:::tip NOTE` and put it at the bottom of the introduction of the `FILL` option.

File: docs/v0.7/en/reference/sql/range.md [hunk 3]
```diff
@@ -60,74 +63,135 @@ ALIGN '5s' BY (host) FILL PREV;
...
+ALIGN '5s';
+```
+:::
 
 Take the following table as an example:
 
 ```sql
 +---------------------+-------+------+
-| ts                  | host  | cpu  |
+| ts                  | host  | val  |
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:96]: If you are using `val` as the column name, please rename all `cpu` column to `val` column in the section of the `FILL` option, including SQLs and results.

File: docs/v0.7/en/reference/sql/range.md [hunk 4]
```diff
@@ -60,74 +63,135 @@ ALIGN '5s' BY (host) FILL PREV;
...
 +---------------------+-------+------+
-| 2023-01-01 08:00:00 | host1 |  4.5 |
-| 2023-01-01 08:00:05 | host1 | NULL |
-| 2023-01-01 08:00:10 | host1 |  6.5 |
+| 1970-01-01 00:00:00 | host1 |    0 |
+| 1970-01-01 00:00:15 | host1 |    6 |
+| 1970-01-01 00:00:00 | host2 |    6 |
+| 1970-01-01 00:00:15 | host2 |   12 |
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:101]: The example should have a row with value `NULL` to show the behavior of the different `FILL` options.

File: docs/v0.7/en/reference/sql/range.md [hunk 5]
```diff
@@ -60,74 +63,135 @@ ALIGN '5s' BY (host) FILL PREV;
 | `LINEAR` | Use [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation) to fill the data. If an integer type is filled with `LINEAR`, the variable type of the column will be implicitly converted to a floating point type during calculation |
 |   `X`    |                                           Fill in a constant, the data type of the constant must be consistent with the variable type of the Range expression                                            |
 
+:::tip NOTE
+Note that if there are multiple Range expressions and the FILL method is specified for only one of them, in order to keep the number of SQL output rows uniform, the other Range expressions will use the FILL NULL method to fill in the missing time slots. So the following two SQL statements are equivalent in output:
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:67]:
```suggestion
If there are multiple Range expressions but only one of them specifies the `Fill` option, the other Range expressions will use the `FILL NULL` method to fill in the missing time slots.
The following two SQL statements are equivalent:
```

**Review by nicecui — COMMENTED** [review]
Please also modify the zh document according to the comments, thank you very much!

File: docs/v0.7/en/reference/sql/range.md (hunk 1 above)
Comment (Taylor-lagrange) [reply docs/v0.7/en/reference/sql/range.md:29]: The `first_value` aggregate function is used to find the first value in an aggregation group. It generally needs to be used with `order by` to specify what the first value is. In range queries, timestamps are widely used as the sorting basis. Therefore, in order to simplify the writing of SQL, the default `order by` parameter of `first_value` is timestamp, and the same for `last_value`.

File: docs/v0.7/en/reference/sql/range.md (hunk 4 above)
Comment (Taylor-lagrange) [reply docs/v0.7/en/reference/sql/range.md:101]: It's the new behavior of FILL option, see pr https://github.com/GreptimeTeam/greptimedb/pull/3489

File: docs/v0.7/en/reference/sql/range.md [hunk 6]
```diff
@@ -308,6 +369,56 @@ Get after running
 +---------------------+-----------------------------------+
 ```
 
+## `ORDER BY` option
+
+Range queries support the use of `order by` expressions within specific aggregate functions to sort data on the same time slot. Taking the `first_value` and `last_value` aggregate functions as an example, users can use `order by` to sort the aggregated data to correctly select the first value and the last value. By default, the `first_value` and `last_value` functions sort data in ascending order using timestamps.
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:374]:
```suggestion
Range queries support the use of `order by` expressions in the parameters of the `first_value` and `last_value` aggregate functions. By default, the data is sorted in ascending order based on the time index column.
```

File: docs/v0.7/en/reference/sql/range.md [hunk 7]
```diff
@@ -308,6 +369,56 @@ Get after running
...
++---------------------+-------+------+-------+
+| ts                  | host  | val  | addon |
++---------------------+-------+------+-------+
+| 1970-01-01 00:00:00 | host1 |    0 |     3 |
+| 1970-01-01 00:00:01 | host1 |    1 |     2 |
+| 1970-01-01 00:00:02 | host1 |    2 |     1 |
++---------------------+-------+------+-------+
+```
+
+Users can not specify the `order by` expression, and the effect is equivalent to using the `ts` column to sort in ascending order.
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:388]:
```suggestion
When the `order by` expression is not specified in the function parameter, the default behavior is to sort the data in ascending order based on the time index column.
```

File: docs/v0.7/en/reference/sql/range.md [hunk 8]
```diff
@@ -308,6 +369,56 @@ Get after running
...
+
+```sql
++---------------------+--------------------------------+-------------------------------+
+| ts                  | FIRST_VALUE(host.val) RANGE 5s | LAST_VALUE(host.val) RANGE 5s |
++---------------------+--------------------------------+-------------------------------+
+| 1970-01-01 00:00:00 |                              0 |                             2 |
++---------------------+--------------------------------+-------------------------------+
+```
+
+Users can also specify their own sorting rules without using the default sorting options, such as using `addon` sorting:
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:406]:
```suggestion
You can specify your own sorting rules. For example, the following SQL sorts the data by the `addon` column in ascending order:
```

File: docs/v0.7/en/reference/sql/range.md [hunk 9]
```diff
@@ -308,6 +369,56 @@ Get after running
...
+Get after query
+
+```sql
++---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
+| ts                  | FIRST_VALUE(host.val) ORDER BY [host.addon ASC NULLS LAST] RANGE 5s | LAST_VALUE(host.val) ORDER BY [host.addon ASC NULLS LAST] RANGE 5s |
++---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
+| 1970-01-01 00:00:00 |                                                                   2 |                                                                  0 |
++---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
+```
+
```
Comment (nicecui) [inline docs/v0.7/en/reference/sql/range.md:421]: What is the behavior of the following SQL:
```
SELECT ts, first_value(val ORDER BY addon ASC) RANGE '5s', last_value(val ORDER BY addon ASC) RANGE '5s' FROM host ALIGN '5s' ORDER BY addon DESC;
```
ORDER BY also added at the end of the SQL and the specified behavior is different from the former.

**Review by nicecui — COMMENTED** [review]
Please also modify the zh document according to the comments, thank you very much

File: docs/v0.7/en/reference/sql/range.md (hunk 9 above)
Comment (nicecui) [reply docs/v0.7/en/reference/sql/range.md:421]: What is the behavior of the SQL that only specified `order by` at the end of the SQL
```
SELECT ts, first_value(val) RANGE '5s', last_value(val) RANGE '5s' FROM host ALIGN '5s' ORDER BY addon DESC;
```

Comment (Taylor-lagrange) [reply docs/v0.7/en/reference/sql/range.md:421]: - `first_value(val ORDER BY addon ASC) RANGE '5s'` means sort data in aggragate group, so we can choose our desire value in `first_value`.
- order by in SQL level meas sort the output of entire output data.

File: docs/v0.7/zh/reference/sql/range.md [hunk 10]
```diff
@@ -304,6 +366,57 @@ FROM host ALIGN '5s' BY ();
 +---------------------+-----------------------------------+
 ```
 
+## 聚合函数中的 `ORDER BY` 选项
+
+Range 查询支持在聚合函数 `first_value` 和 `last_value` 中使用 `order by` 表达式，默认情况下，聚合函数使用时间列升序排列数据。
```
Comment (nicecui) [inline docs/v0.7/zh/reference/sql/range.md:371]:
```suggestion
Range 查询支持在聚合函数 `first_value` 和 `last_value` 中使用 `order by` 表达式，默认情况下，聚合函数使用时间索引列升序排列数据。
```

**Review by nicecui — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Review by tisonkun — COMMENTED** [review]
@Taylor-lagrange @nicecui we should update the `FILL` table to avoid

```
|  `NULL`  |                                                                                Fill directly with `NULL` (default method)                                                                                |
```

Since the default is now "NO FILL"

**Pull request merged**